### PR TITLE
Core / Entitiy

### PR DESCRIPTION
**Pull request by dgkncelik**
Methods added for getting component / participant objects from repositories.

### CODE REVIEW

**Review by karakayasemi — APPROVED** [review]
Generally, I am okay with commits. Just CIs should pass before merge. There are compilation errors.

**Comment by karakayasemi** [comment]
I will add some tests and fix CI

**Comment by karakayasemi** [comment]
We need to add more tests to make codecov happy.

File: src/main/java/edu/itu/cavabunga/core/factory/ComponentFactory.java [hunk 1]
```diff
@@ -2,11 +2,75 @@
...
+import java.util.Date;
 
 @org.springframework.stereotype.Component
 public class ComponentFactory {
 
-    public Component createComponent(ComponentType componentType) {
-        return componentType.create();
+    public Component createComponent(ComponentType componentType) throws IcalStorageException {
+        Component result;
+        switch (componentType){
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/core/factory/ComponentFactory.java]: There are duplications in here, we are calling exactly same code block

File: src/main/java/edu/itu/cavabunga/core/factory/ComponentFactory.java [hunk 2]
```diff
@@ -2,11 +2,75 @@
 
 import edu.itu.cavabunga.core.entity.Component;
 import edu.itu.cavabunga.core.entity.component.ComponentType;
+import edu.itu.cavabunga.core.exception.IcalStorageException;
+
+import java.util.Date;
 
 @org.springframework.stereotype.Component
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/core/factory/ComponentFactory.java:9]: is it really necessary?

Comment (dgkncelik) [reply src/main/java/edu/itu/cavabunga/core/factory/ComponentFactory.java:9]: How else we store date data format ? string ?

Comment (karakayasemi) [reply src/main/java/edu/itu/cavabunga/core/factory/ComponentFactory.java:9]: I was talking about @org.springframework.stereotype.Component . Whatever:) it is necessary

**Comment by karakayasemi** [comment]
@dgkncelik review my last commit, is it okay for you?

**Comment by karakayasemi** [comment]
By the way, I did not like the TestNG data provider, autowired annotation does not work easily with it.

**Comment by karakayasemi** [comment]
yey, scrutinizer likes me.
image

**Pull request merged**